gradio/components.py

```diff
@@ -3051,16 +3051,22 @@ def postprocess(
         if y is None:
             return self.postprocess(self.empty_input)
         if isinstance(y, dict):
+            y["headers"] = y["headers"][: self.max_cols]
+            y["data"] = list(
+                map(lambda row: row[: self.max_cols], y["data"][: self.max_rows])
+            )
             return y
         if isinstance(y, str):
-            dataframe = pd.read_csv(y)
+            dataframe = pd.read_csv(y, nrows=self.max_rows)
+            dataframe = dataframe.loc[:, dataframe.columns[: self.max_cols]]
```

Comment:
.loc is ok here since we're only indexing the columns by name

Reply:
Thanks for pointing this out, will update it to iloc as well for consistency, should be more efficient too